stm32H573 fails to build spi_ll_stm32.c  with manual cache coherency

When building the tests/drivers/spi/spi_loopback/ for the stm32H573i_dk target board
if the  **SPI_STM32_MANUAL_CACHE_COHERENCY_REQUIRED** is set, the build fails due to 
```
~/drivers/spi/spi_ll_stm32.c:173: undefined reference to `arch_dcache_flush_range'
collect2.exe: error: ld returned 1 exit status
```

The SPI_STM32_MANUAL_CACHE_COHERENCY_REQUIRED is for "manual cache coherency management is only done on dummy_rx_tx_buffer" 
**for the  drivers.spi.stm32_spi_dma.loopback testcase in the tests/drivers/spi/spi_loopback

This error seems to come after the merge of the ¨PR # https://github.com/zephyrproject-rtos/zephyr/pull/88585


**Impact**
CI failure 


**Environment (please complete the following information):**
 - Toolchain SDK 0.17.0
 - Commit SHA 3e09a40de5e80c3972a856a9b37479e8f3bc54d6

--- a/drivers/spi/spi_ll_stm32.c
+++ b/drivers/spi/spi_ll_stm32.c
@@ -15,6 +15,7 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
 #include <soc.h>
 #include <stm32_ll_spi.h>
 #include <errno.h>
+#include <zephyr/cache.h>
 #include <zephyr/drivers/spi.h>
 #include <zephyr/drivers/spi/rtio.h>
 #include <zephyr/drivers/pinctrl.h>
@@ -168,7 +169,7 @@ static int spi_stm32_dma_tx_load(const struct device *dev, const uint8_t *buf,
 		/* if tx buff is null, then sends NOP on the line. */
 		dummy_rx_tx_buffer = 0;
 #if SPI_STM32_MANUAL_CACHE_COHERENCY_REQUIRED
-		arch_dcache_flush_range((void *)&dummy_rx_tx_buffer, sizeof(uint32_t));
+		sys_cache_data_flush_range((void *)&dummy_rx_tx_buffer, sizeof(uint32_t));
 #endif /* SPI_STM32_MANUAL_CACHE_COHERENCY_REQUIRED */
 		blk_cfg->source_address = (uint32_t)&dummy_rx_tx_buffer;
 		blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;

